field/src/batch_inverse.rs

```diff
@@ -24,22 +45,10 @@ pub fn batch_multiplicative_inverse<F: Field>(x: &[F]) -> Vec<F> {
     // Handle special cases. Paradoxically, below is repetitive but concise.
     // The branches should be very predictable.
     let n = x.len();
-    if n == 0 {
-        return Vec::new();
-    } else if n == 1 {
-        return vec![x[0].inverse()];
-    } else if n == 2 {
-        let x01 = x[0] * x[1];
-        let x01inv = x01.inverse();
-        return vec![x01inv * x[1], x01inv * x[0]];
-    } else if n == 3 {
-        let x01 = x[0] * x[1];
-        let x012 = x01 * x[2];
-        let x012inv = x012.inverse();
-        let x01inv = x012inv * x[2];
-        return vec![x01inv * x[1], x01inv * x[0], x012inv * x01];
```

Comment:
Given the code is already written, we might as well copy it into batch_multiplicative_inverse_small.

Reply:
It gets slightly more verbose when writing to a result buffer, and I was thinking it'd be nice to have a small separate function, ideally for arbitrary `n` just so no context/precondition is needed.

Maybe it can be a naive (fully serial) implementation of Montgomery's trick? It's slightly less code than the four cases, probably a bit more control-flow overhead but it shouldn't affect us. Might also be convenient as a baseline for benching other implementations.

Reply:
Hm, we could almost rewrite the normal "chunked" batch inverse impl to use (a slight generalization of) this naive impl. I.e. we could call it with `FieldArray`, and gave `FieldArray` an inverse method which triggered a small second batch inversion. There are a few complications, like handing a final smaller chunk, but I'll see if there's a nice way to do it.

Reply:
Created #484 for this optional variation